src/platform/Zephyr/BLEManagerImpl.cpp

```diff
@@ -273,6 +287,12 @@ CHIP_ERROR BLEManagerImpl::StartAdvertising(void)
     chip::Encoding::LittleEndian::Put16(serviceData.uuid, UUID16_CHIPoBLEService.val);
     ReturnErrorOnFailure(ConfigurationMgr().GetBLEDeviceIdentificationInfo(serviceData.deviceIdInfo));
 
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+    ReturnErrorOnFailure(PrepareC3CharData());
+    // Increment life time counter to protect against long-term tracking of rotating device ID.
+    ReturnErrorOnFailure(ConfigurationMgr().IncrementLifetimeCounter());
```

Comment:
Maybe it would be better to increment this in `StopAdvertising` so that rotating device ID stays unchanged in case a device maker would like to expose it during the advertising through another channel?